### DISCUSSION TITLE
The heckin chonker update

### DISCUSSION THREAD

**Discussion by cryptofyre** [ciderapp/Apple-Music-Electron#107] [General]
![image](https://user-images.githubusercontent.com/33162551/124037921-f0670480-d9c5-11eb-82ec-a8ef56c01c84.png)
preview image of the new Dracula theme on v2.3.0
# The heckin' chonker update
### This update contains a whole motherload of internal changes and user customization updates.
The big change to this update is the ability to change your user config without having to recompile the application.

On Windows:
You can find your config in the Apple Music folder in your documents folder.

On Linux:
You can find your config in the `.config/'Apple Music'` directory

On macOS:
You can find your config in the `Library/Application Support/Apple Music/` directory.

## Changelog:
* Customizable user configuration
* Added a few community themes
* Edited Dracula to look more visually appealing
* Split index.js into multiple function files and other files to make things easier to edit
* Optimized and changed a ton of features
* pacman packages can be built now by running `yarn dist -l pacman`
* Made the thumbbar icons more visible on certain color preferences (dark or light)
* System colors now affect the application theme
* and more.

<hr /><em>This discussion was created from the release <a href='https://github.com/cryptofyre/Apple-Music-Electron/releases/tag/v2.3.0'>The heckin chonker update</a>.</em>

**Comment by eliasib13** [comment]
I downloaded the release 2.1.0 like 10 minutes before 2.3.0 was released, and I was missing the config.json 😂 Thanks for the update, the app looks quite good!

**Reply by cryptofyre** [reply]
haha

**Comment by aliquo93** [comment]
How can I move minimize, maximize and close buttons to left? Or how can I force the app to use the ones form my system? Anyway, very good app!

**Reply by cryptofyre** [reply]
You can change this by editing your config file in C:/youruser/Documents/Apple Music/config.json and setting macOSWindowForceRight to false.

**Reply by aliquo93** [reply]
Worked! Thank you very much!

**Comment by aliquo93** [comment]
Could you also tell me how to change the tray icon to a custom one? Thanks in advance!

**Reply by cryptofyre** [reply]
Recompiling would be needed but you can change the icon in the repos resources/icons/ folder. Make sure to change icon.ico and icon.png as icon.icns is purely for macOS and icon.ico being for Windows.

**Reply by aliquo93** [reply]
Thanks!

**Comment by TheBigChris** [comment]
Hey there! First-time user (this app is wonderful! Really wanted a way to use apple music on PC without having to resort to iTunes or the web client) Downloaded, installed, and everything worked well. Then I signed in with my Apple ID. Nonstop error message, unresolved by re-install:

"Apple Music is not available in your country or region."

Now I know for a fact that this is not correct, as I live on the US west coast. I'm not using a VPN or anything and iTunes and the Web app are both happy enough to stream apple music. 

Any thoughts on what I should try to fix this? The only action I took recently of note was authorizing this computer on iTunes to download music, though I think that might be unrelated. 
![Screenshot (102)](https://user-images.githubusercontent.com/87000739/124548977-59de7d00-dde3-11eb-87b8-4665f47df20f.png)

**Reply by TheBigChris** [reply]
Even tried creating a new Apple ID inside of the client, ran into the same error.

**Reply by Nosh118** [reply]
Have you tried disabling CORS in the config file? It should be located in an Apple Music electron folder in your documents. Remember to quit AME from the taskbar as well to ensure the config changes take place.

**Reply by coredev-uk** [reply]
In the new release, we have added an auth mode. Try downloading the new beta and enabling it. (You may need to delete your current configuration as issues could occur).  Make sure to disable the option after the login has been completed.

**Reply by nigelonius** [reply]
I'm on version 2.4 and have CORS disabled but still running into this issue. Any other fix?

**Reply by rafalozan0** [reply]
Hi! i just discovered this awesome app, but im having the same issue, the app say that apple music is not available in my country :(, i tried disabling the CORS but still not working

**Reply by quacksire** [reply]
> Hi! i just discovered this awesome app, but im having the same issue, the app say that apple music is not available in my country :(, i tried disabling the CORS but still not working

This is fixed in the 2.4.0 release, which you can get a beta release in the the Releases tab

**Reply by nigelonius** [reply]
> > Hi! i just discovered this awesome app, but im having the same issue, the app say that apple music is not available in my country :(, i tried disabling the CORS but still not working
> 
> This is fixed in the 2.4.0 release, which you can get a beta release in the the Releases tab

This issue still exists on 2.4.0

**Reply by nigelonius** [reply]
So I seem to have found a fix. Once you open it for the first time (I'm on 2.4) press Ctrl+Alt+S to enter settings. Click authentication mode under general settings. Then head over to advance settings on the left navigation pane and under ForceApplicationRegion, select your region. Quit the app completely (including shutting it down in the system tray) and then run it again and log in. This should work for that "Apple Music unavailable in your region error"...it did for me.

**Reply by rafalozan0** [reply]
![image](https://user-images.githubusercontent.com/25993201/125243605-eee5e800-e2b3-11eb-840c-599d227a70f8.png)

yeeeaahhhhh!! @child-duckling  @nigelonius  thank you so much guys, it worked! the app is amazing 🤯🤯

**Reply by TheBigChris** [reply]
Working now @nigelonius @coredev-uk @child-duckling 

!!!

Thank you all so much for your help! It's always difficult to get anything "apple" to work where it does not want to if you know what I mean ;)

Can't really express how much of a fan of your work I am. You guys rock. Keep it up!

**Reply by ghost** [reply]
> 
> 
> So I seem to have found a fix. Once you open it for the first time (I'm on 2.4) press Cyril+Alt+Windows key to enter settings. Click authentication mode under general settings. Then head over to advance settings on the left navigation pane and under ForceApplicationRegion, select your region. Quite the app completely (including shutting it down in the system tray) and then run it again and log in. This should work for that "Apple Music unavailable in your region error"...it did for me.

Ctrl+Alt+Windows does nothing for me, I can't even get the login once.

**Comment by nigelonius** [comment]
They want it that way so that we feel obligated to purchase their hardware
to get the full functionality of their services. I'm glad you got it to
work for you because I know how frustrating it was when it wasn't working
for me.

**Comment by ghost** [comment]
Sorry for the ignorant question, but when can we expect the newer version? I haven't been able to use it because of the 'Can't login' bug.

**Reply by cryptofyre** [reply]
Version 2.4.0 is coming out soon and should fix all your login issues as it has appeared to do so for everyone so far.

**Comment by nigelonius** [comment]
Sorry, it’s actually CTRL+Alt+S
________________________________

**Comment by nigelonius** [comment]
I haven't used this software in 4 months, and I thought I would try it again now. Y'all!!! I bow down to you! You have done the most👏. It looks A M A Z I N G! It's so stable, I have made it my fulltime player (goodbye Spotify). Sincerely, I would like to thank every single one of you that has put their arduous work into producing something this premium. You have put Apple to shame. Keep up the extraordinary work y'all!
